--- accel/hvf/hvf-all.c.orig	2024-12-11 07:46:36.000000000 +0800
+++ accel/hvf/hvf-all.c	2024-12-14 08:15:06.000000000 +0800
@@ -23,7 +23,10 @@
     case HV_NO_RESOURCES: return "HV_NO_RESOURCES";
     case HV_NO_DEVICE:    return "HV_NO_DEVICE";
     case HV_UNSUPPORTED:  return "HV_UNSUPPORTED";
+#if defined(MAC_OS_VERSION_11_0) && \
+    MAC_OS_X_VERSION_MIN_REQUIRED >= MAC_OS_VERSION_11_0
     case HV_DENIED:       return "HV_DENIED";
+#endif
     default:              return "[unknown hv_return value]";
     }
 }
