--- include/qemu/osdep.h.orig	2024-12-11 07:46:36.000000000 +0800
+++ include/qemu/osdep.h	2024-12-14 06:30:27.000000000 +0800
@@ -801,7 +801,8 @@
  * Toggle write/execute on the pages marked MAP_JIT
  * for the current thread.
  */
-#ifdef __APPLE__
+#if defined(MAC_OS_VERSION_11_0) && \
+    MAC_OS_X_VERSION_MIN_REQUIRED >= MAC_OS_VERSION_11_0
 static inline void qemu_thread_jit_execute(void)
 {
     pthread_jit_write_protect_np(true);
